gluten-core/src/main/resources/substrait/proto/substrait/algebra.proto

```diff
@@ -319,6 +319,15 @@ message WindowRel {
   }
 }
 
+message WindowGroupLimitRel {
```

Comment:
@ayushi-agarwal It seems that the only difference between WindowGroupLimitRel and WindowRel is the "limit" parameter? If that's the case, can we simply use WindowRel and pass the "limit" parameter through advanced_extension? This would prevent the introduction of a new operator in the Substrait code.

Reply:
@JkSelf For now we are only supporting row number but later when we support rank and dense rank, then I was thinking that we would need an extra parameter for passing the function name, so I created the new operator. Shall that info also be passed in advanced_extension?

Reply:
@ayushi-agarwal Yes, it can be included in the advanced_extension. However, it appears that the WindowGroupLimitRel is intended to map to the TopNRowNumber operator, not the Window operator. Therefore, it make sense to introduce a new operator in Substrait for this purpose. @zhztheplayer Do you have any input?

Reply:
> Therefore, it make sense to introduce a new operator in Substrait for this purpose.

I am inclined to this approach, like adding a `TopNRowNumberRel` or something in proto. 1:1 mapping from Velox's operator to Substrait / Gluten operator is definitely encouraged, like discussed in https://github.com/apache/incubator-gluten/pull/5409#discussion_r1566884037.

Reply:
To make things fancier, we can even add a `TopNRowNumberExec` in `backends-velox` module then use a rule to convert Spark's `WindowGroupLimitExec` conditionally to `TopNRowNumberExec`. That would be the best method to solve this kind of operator-mapping problems.

However if you think `WindowGroupLimitExec` and `TopNRowNumberRel` doesn't have much semantic difference apart of the window function types they support, then we can do it without rules and it's also fine.

Reply:
I kept it as WindowGroupLimitRel and not TopNRowNumberRel so that later we can use the same for rank and denseRank.